### PR TITLE
Updates to cmake files:

### PR DESCRIPTION
**Pull request by mk01**
- new FindEmbedded - it looks for presence of RPI or IMX6
- if Embedded found:
    - OpenGL find is skipped (this might find libGL.so on IMX6 otherwise)
    - pkgconfig is skipped (this might find Mesa libs on RPI if installed)

(that means, for IMX6 GLES2 will be used, on RPI as well with preference
of /opt/vc if found)

### CODE REVIEW

File: FindEMBEDDED.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# - Try to find embedded platforms (RPI/IMX6)
+# Once done this will define
+#
+#  EMBEDDED_FOUND        - system is RPI / IMX6 and we most probably want to compile for GLES2 support
+#                          (don't configure for OpenGL)
+
+find_path(EMBEDDED_FOUND NAMES linux/imxfb.h bcm_host.h HINTS /opt/vc/include)
```
Comment (wsnipex) [inline FindEMBEDDED.cmake]: please use PATHS instead of HINTS. Also in FindOpenGLES2

**Comment by wsnipex** [comment]
Looks ok to me, but I'm unsure about skipping pkgconfig.
@popcornmix ping

**Comment by mk01** [comment]
@wsnipex 

i was thinking of that too - and specially checked. at least in "firmware" there is no .pc supplied at all. 
but don't know if there is/isn't a special .pc supplied by any of raspbian packages... that has @popcornmix to confirm or not

**Comment by mk01** [comment]
@wsnipex 

if will retest with PATHS. just had the feeling PATHS comes during cmake checks after system standard paths what would link RPI against mesa libraries if installed. 
will check.

File: FindEMBEDDED.cmake (hunk 1 above)
Comment (FernetMenta) [inline FindEMBEDDED.cmake]: not sure if "most probably" is the way to go. If I read this right it breaks OpenGL on a system that can cross compile for rbp, right?

Comment (wsnipex) [inline FindEMBEDDED.cmake]: yes, thats most probably correct, good catch. Maybe also checking if target arch(CMAKE_SYSTEM_PROCESSOR) is arm might be enough.

Comment (wsnipex) [inline FindEMBEDDED.cmake]: confirmed, it breaks in the case @FernetMenta outlined.
try if
>if(CMAKE_SYSTEM_PROCESSOR MATCHES "arm")
works for you

**Comment by mk01** [comment]
@wsnipex 

I couldn't force findlibary() by using PATHS to prefer /opt/vc so I changed logic to prepend EMBEDDED_FOUND to CMAKE_PREFIX_PATH (like Toolchain_binaddons.cmake.is doing)

I retested imx6/rpi builds and pushed.

**Comment by wsnipex** [comment]
you still break native builds on systems that can cross compile embedded, see comments above.

**Comment by mk01** [comment]
@wsnipex 

missed that comments on diff initially. now tested with rpi/imx6 and native linux/x86 compilation with /opt/vc copied to root.

**Comment by wsnipex** [comment]
@stefansaraev mind testing this on OE?
@FernetMenta your button

**Comment by stefansaraev** [comment]
before this pr: http://sprunge.us/dVAd
after this pr: http://sprunge.us/ZBHS

CMakeOutput.log: http://sprunge.us/DPjU

no clue what's going on so I can only test but not actively help. cmake is like speaking chinese to me ;)

File: FindEMBEDDED.cmake [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# - Try to find embedded platforms (RPI/IMX6)
+# Once done this will define
+#
+#  EMBEDDED_FOUND        - system is RPI / IMX6 and we most probably want to compile for GLES2 support
+#                          (don't configure for OpenGL)
+
+string(REGEX MATCH ^arm TARGET_ARCH_ARM ${CMAKE_SYSTEM_PROCESSOR})
```
Comment (wsnipex) [inline FindEMBEDDED.cmake]: string(REGEX MATCH "^arm" TARGET_ARCH_ARM "${CMAKE_SYSTEM_PROCESSOR}")

Comment (stefansaraev) [inline FindEMBEDDED.cmake]: ^^ fixes the build failure I reported above. thanks.

**Comment by FernetMenta** [comment]
@wsnipex I guess yo don't want me to press the button right now, do you?

**Comment by wsnipex** [comment]
yes, the fix I mentioned above is needed first.

**Comment by mk01** [comment]
@wsnipex 

changed accordingly

**Pull request merged**